runtime/src/comm/gasnet/comm-gasnet.c

```diff
@@ -340,6 +390,32 @@ static void AM_bcast_seginfo(gasnet_token_t token, void *buf, size_t nbytes) {
   bcast_seginfo_done = 1;
 }
 
+// Put from arg->src (which is local to the AM handler) back to
+// arg->dst (which is local to the caller of this AM).
+// nbytes must be < gasnet_AMMaxLongReply.
```

Comment:
It's not checked - it's just written to meet that requirement.  The comment around line 1023 is describing how the loop calling AM_reply_put (called DO_REPLY_PUT when it calls it) is using chunks of size <gasnet_AMMaxLongReply.
Would you like me to add an assert here?